Add Data Model Updates

dan-tang-ssd opened the pull request:
This PR is submitted to fix https://github.com/stats4sd/ae-policy-tracking-tool/issues/8

It is not yet ready for review. It is submitted for progress update.

---

It contains below changes:
 - [x] Issues occurred when setting up local development env
 - [x] Policy model updates
 - [x] Rename Policy model to PolicyDocument model, update model, relationships, filament resources, blade view files
 - [x] Add a new ENUM class for text direction
 - [x] Add Highlight model and relationship
 - [x] Policy document resource > file upload components, mock up code to illustrate how to split uploaded files into two different file upload components
 - [x] Policy document resource > file upload components, revise to check if there are highlights for uploaded files

dan-tang-ssd committed:
issues when setting up local development env

dan-tang-ssd commented:
It took me a little longer to setup local development env.

I setup this application as usual: 
 - git clone
 - git submodule init
 - git submodule update
 - composer install
 - npm install
 - php artisan key:generate
 - php artisan migrate:fresh --seed

I got error when running seeder file, it tries to insert record to a non-existed "teams" table.

SQLSTATE[42S02]: Base table or view not found: 1146 Table 'ae_policy_tracking.teams' doesn't exist (Connection: mysql, SQL: insert into `teams` (`country_id`, `status`, `finalised_at`, `title`, `updated_at`, `created_at`) values (1, In Progress, ?, Uganda 2025, 2025-10-13 16:16:58, 2025-10-13 16:16:58))

---

I wondered if we need to run the install script for package filament-team-management

php artisan filament-team-management:install

It publishes migration files for table "teams" and "team_members". But they do not help as "teams" table does not have column "country_id".

I checked against staging db, realised that the seeder file should insert records to "assessments" table instead.

I have removed the two published migration files afterwards.

Instead of fixing this issue, I decided to clone staging db to local db to make it start running first.

---

In my .env file, I specify the application URL as http://ae-policy-tool.test.

`APP_URL=http://ae-policy-tool.test`

When I visit http://ae-policy-tool.test in browser, it returns error 404.

When I visit http://ae-policy-tracking-tool.test, it shows app panel login page.

It looks like it does not follow the URL defined in APP_URL config item.
By the way, I have updated my APP_URL config item to http://ae-policy-tracking-tool.test

`APP_URL=http://ae-policy-tracking-tool.test`

---

For account "support@stats4sd.org", I reset the login password in backend database.
When I login app panel, error occurred again for complaining table "teams" is missing.

It looks like the application does not know the correct table name is "assessments" in my local env.
I am using PHP 8.4.12, Laravel 11.0.

Question: Is there anything I need to config or setup at my side to make it work?

```
dan@dan-XPS-9320:~$ php -v
PHP 8.4.12 (cli) (built: Aug 29 2025 06:48:12) (NTS)
Copyright (c) The PHP Group
Built by Debian
Zend Engine v4.4.12, Copyright (c) Zend Technologies
    with Zend OPcache v8.4.12, Copyright (c), by Zend Technologies
dan@dan-XPS-9320:~$ 
```

```
composer.json

"laravel/framework": "^11.0",

```

---

I ignore the error and visit admin panel by changing URL directly. It works.

Filament resources under Lookup Lists group are working fine.

In User Management group, error occurred as Assessments and Users resources are looking for the missing "teams" table.

---

Screen shots:

App panel login and error

<img width="1920" height="982" alt="image" src="https://github.com/user-attachments/assets/a322549d-c38c-4836-aaa7-5e3207db1456" />

<img width="1920" height="982" alt="image" src="https://github.com/user-attachments/assets/d470d844-3f8b-4609-9def-3cdb7301200c" />

---

Access admin panel directly

<img width="1920" height="982" alt="image" src="https://github.com/user-attachments/assets/6eef69fd-79b3-4f98-a0b2-5ae2a638ec79" />

Error occurred when access Assessments resource or Users resource

<img width="1920" height="982" alt="image" src="https://github.com/user-attachments/assets/3a5ae454-793e-48a3-acd4-709a33a687df" />

dan-tang-ssd commented:
I will start working on updating the models first, as it should not be affected by the mentioned error.

dan-tang-ssd committed:
add column policies.content, policies.text_direction; add PHP enum class TextDirection; add casts in Policy model

dan-tang-ssd committed:
add migration files to create highlights table and highlight_priority_action table; create model Highlight; add many-to-many relationship in Highlight model and PriorityAction model

dan-tang-ssd committed:
add relationships in models

dan-tang-ssd commented:
To continue development work, I have created "teams" table by this way:
1. export "assessments" table 
2. in mysqldump file, changed table name from "assessments" to "teams"
3. import "teams" table

Now we have "teams" table, app panel works well.

---

Screen shots:

<img width="1920" height="982" alt="image" src="https://github.com/user-attachments/assets/a362ed68-8672-4c87-9d91-6d91aab2bf84" />

dan-tang-ssd committed:
rename Policy model as PolicyDocument model; rename relationships; update filament resource and related code

dan-tang-ssd committed:
mock up code to illutstrate how to split uploaded files into two different file upload components

dan-tang-ssd committed:
add column highlights.media_id

dan-tang-ssd committed:
Policy document resource > file upload components, revise to check if there are highlights for uploaded file

dan-tang-ssd commented:
I have revised program file to check if there is any highlight for uploaded policy document file.
I manually added some highlights records in backend database for testing.

---

Screen shots:

<img width="1920" height="982" alt="image" src="https://github.com/user-attachments/assets/6c0cf5de-c1a7-45ec-8999-e0c250979e28" />


<img width="715" height="149" alt="image" src="https://github.com/user-attachments/assets/0249a024-cc12-4a02-82b7-ad259eea7674" />

dan-tang-ssd commented:
I think it is now ready for review.

dave-mills committed:
update .env.example with correct TEAM_MANAGEMENT vars

dave-mills commented:
> When I login app panel, error occurred again for complaining table "teams" is missing.

> It looks like the application does not know the correct table name is "assessments" in my local env.

Ah, looks like I hadn't updated the .env.example with the correct FILAMENT_TEAM_MANAGEMENT_*** variables. We can actaully use any model / table as the "team", "user" or "program" table, and can tell the team management package which models to use through the .env variables. I've just updated it, so the seeder / app should work for you locally if you update the .env file.

dave-mills reviewed (approved):
This looks good - I'm happy to merge this in, and then use it as the base to bring in the highlighting tool I've been exploring. 

@dan-tang-ssd for the extra policy document multi-uploader page, I suggest that becomes a new branch, so we can both have these updates.

dan-tang-ssd commented:
> > When I login app panel, error occurred again for complaining table "teams" is missing.
> 
> > It looks like the application does not know the correct table name is "assessments" in my local env.
> 
> Ah, looks like I hadn't updated the .env.example with the correct FILAMENT_TEAM_MANAGEMENT_*** variables. We can actaully use any model / table as the "team", "user" or "program" table, and can tell the team management package which models to use through the .env variables. I've just updated it, so the seeder / app should work for you locally if you update the .env file.

Thank you Dave for the fix.

I pulled latest code from dev, updated my .env file by referring to .env.example. I have also deleted teams table in my local database. 

I can login successfully. It works properly now.
